kv/kvserver: TestGossipHandlesReplacedNode failed

kv/kvserver.TestGossipHandlesReplacedNode [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/15539304?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/15539304?buildTab=artifacts#/) on release-23.2 @ [fb0f1a4da32de07966847a0c9be39fdad77a2a76](https://github.com/cockroachdb/cockroach/commits/fb0f1a4da32de07966847a0c9be39fdad77a2a76):


```
=== RUN   TestGossipHandlesReplacedNode
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestGossipHandlesReplacedNode910954614
    test_log_scope.go:81: use -show-logs to present logs inline
    testcluster.go:679: timed out waiting for 4 store descriptors: map[1:{} 2:{} 3:{}]
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestGossipHandlesReplacedNode910954614
--- FAIL: TestGossipHandlesReplacedNode (60.85s)
```

Parameters:
 - <code>TAGS=bazel,gss,deadlock</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #122735 kv/kvserver: TestGossipHandlesReplacedNode failed [A-testing C-bug C-test-failure O-robot P-3 T-kv branch-master]
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGossipHandlesReplacedNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-39317

--- a/pkg/kv/kvserver/gossip_test.go
+++ b/pkg/kv/kvserver/gossip_test.go
@@ -146,6 +146,7 @@ func TestGossipHandlesReplacedNode(t *testing.T) {
 
 	// As of Nov 2018 it takes 3.6s.
 	skip.UnderShort(t)
+	skip.UnderDeadlock(t)
 	ctx := context.Background()
 
 	// Shorten the raft tick interval and election timeout to make range leases

